firedrake/tsfc_interface.py

```diff
@@ -84,7 +84,7 @@ def _cache_store(cls, key, val):
             # No need for a barrier after this, since non root
             # processes will never race on this file.
             with gzip.open(tempfile, 'wb') as f:
-                cPickle.dump(val, f)
+                cPickle.dump(val, f, 0)
```

Comment:
Default protocol number is 0 in Python 2, and 3 in Python 3, and I forgot what was actually wrong with this one...